### ISSUE TITLE
[babel-plugin-flow-runtime] cannot read .name of null 

### ISSUE THREAD

**Issue by aretecode** [codemix/flow-runtime#14]
using default plugin config (`flow-runtime`), or as object with the defaults explicitly specified (`'assert': true, 'decorate': true`...)

```js
Cannot read property 'name' of null
   at converters.FunctionTypeParam (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:539:29)
   at convert (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:38:12)
   at /project/node_modules/babel-plugin-flow-runtime/lib/convert.js:512:12
   at Array.map (native)
   at converters.FunctionTypeAnnotation (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:511:62)
   at convert (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:38:12)
   at converters.ObjectTypeProperty (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:498:15)
   at convert (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:38:12)
   at /project/node_modules/babel-plugin-flow-runtime/lib/convert.js:488:12
   at Array.map (native)
   at converters.ObjectTypeAnnotation (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:487:114)
   at convert (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:38:12)
   at converters.TypeAlias (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:304:14)
   at convert (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:38:12)
   at TypeAlias (/project/node_modules/babel-plugin-flow-runtime/lib/transformVisitors.js:69:47)
   at NodePath._call (/project/node_modules/babel-traverse/lib/path/context.js:76:18)
   at NodePath.call (/project/node_modules/babel-traverse/lib/path/context.js:48:17)
   at NodePath.visit (/project/node_modules/babel-traverse/lib/path/context.js:105:12)
   at TraversalContext.visitQueue (/project/node_modules/babel-traverse/lib/context.js:150:16)
   at TraversalContext.visitMultiple (/project/node_modules/babel-traverse/lib/context.js:103:17)
   at TraversalContext.visit (/project/node_modules/babel-traverse/lib/context.js:190:19)
   at Function.traverse.node (/project/node_modules/babel-traverse/lib/index.js:114:17)
   at convert (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:40:11)
   at /project/node_modules/babel-plugin-flow-runtime/lib/convert.js:512:12
   at Array.map (native)
   at converters.FunctionTypeAnnotation (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:511:62)
   at convert (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:38:12)
   at converters.ObjectTypeProperty (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:498:15)
   at convert (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:38:12)
   at /project/node_modules/babel-plugin-flow-runtime/lib/convert.js:488:12
   at Array.map (native)
   at converters.ObjectTypeAnnotation (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:487:114)
   at convert (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:38:12)
   at converters.TypeAlias (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:304:14)
   at convert (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:38:12)
   at TypeAlias (/project/node_modules/babel-plugin-flow-runtime/lib/transformVisitors.js:69:47)
   at NodePath._call (/project/node_modules/babel-traverse/lib/path/context.js:76:18)
   at NodePath.call (/project/node_modules/babel-traverse/lib/path/context.js:48:17)
   at NodePath.visit (/project/node_modules/babel-traverse/lib/path/context.js:105:12)
   at TraversalContext.visitQueue (/project/node_modules/babel-traverse/lib/context.js:150:16)
   at TraversalContext.visitMultiple (/project/node_modules/babel-traverse/lib/context.js:103:17)
   at TraversalContext.visit (/project/node_modules/babel-traverse/lib/context.js:190:19)
   at Function.traverse.node (/project/node_modules/babel-traverse/lib/index.js:114:17)
   at traverse (/project/node_modules/babel-traverse/lib/index.js:79:12)
   at NodePath.traverse (/project/node_modules/babel-traverse/lib/path/index.js:144:25)
   at convert (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:40:11)
   at converters.ObjectTypeProperty (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:498:15)
   at convert (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:38:12)
   at /project/node_modules/babel-plugin-flow-runtime/lib/convert.js:488:12
   at Array.map (native)
   at converters.ObjectTypeAnnotation (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:487:114)
   at convert (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:38:12)
   at converters.TypeAlias (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:304:14)
   at convert (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:38:12)
   at TypeAlias (/project/node_modules/babel-plugin-flow-runtime/lib/transformVisitors.js:69:47)
   at NodePath._call (/project/node_modules/babel-traverse/lib/path/context.js:76:18)
   at NodePath.call (/project/node_modules/babel-traverse/lib/path/context.js:48:17)
   at NodePath.visit (/project/node_modules/babel-traverse/lib/path/context.js:105:12)
   at TraversalContext.visitQueue (/project/node_modules/babel-traverse/lib/context.js:150:16)
   at TraversalContext.visitMultiple (/project/node_modules/babel-traverse/lib/context.js:103:17)
   at TraversalContext.visit (/project/node_modules/babel-traverse/lib/context.js:190:19)
   at Function.traverse.node (/project/node_modules/babel-traverse/lib/index.js:114:17)
   at traverse (/project/node_modules/babel-traverse/lib/index.js:79:12)
   at NodePath.traverse (/project/node_modules/babel-traverse/lib/path/index.js:144:25)
   at PluginPass.Program (/project/node_modules/babel-plugin-flow-runtime/lib/index.js:20:14)
   at newFn (/project/node_modules/babel-traverse/lib/visitors.js:276:21)
   at NodePath._call (/project/node_modules/babel-traverse/lib/path/context.js:76:18)
   at convert (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:40:11)
   at /project/node_modules/babel-plugin-flow-runtime/lib/convert.js:488:12
   at Array.map (native)
   at converters.ObjectTypeAnnotation (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:487:114)
   at convert (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:38:12)
   at converters.TypeAlias (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:304:14)
   at convert (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:38:12)
   at TypeAlias (/project/node_modules/babel-plugin-flow-runtime/lib/transformVisitors.js:69:47)
   at NodePath._call (/project/node_modules/babel-traverse/lib/path/context.js:76:18)
   at NodePath.call (/project/node_modules/babel-traverse/lib/path/context.js:48:17)
   at NodePath.visit (/project/node_modules/babel-traverse/lib/path/context.js:105:12)
   at TraversalContext.visitQueue (/project/node_modules/babel-traverse/lib/context.js:150:16)
   at TraversalContext.visitMultiple (/project/node_modules/babel-traverse/lib/context.js:103:17)
   at TraversalContext.visit (/project/node_modules/babel-traverse/lib/context.js:190:19)
   at Function.traverse.node (/project/node_modules/babel-traverse/lib/index.js:114:17)
   at traverse (/project/node_modules/babel-traverse/lib/index.js:79:12)
   at NodePath.traverse (/project/node_modules/babel-traverse/lib/path/index.js:144:25)
   at PluginPass.Program (/project/node_modules/babel-plugin-flow-runtime/lib/index.js:20:14)
   at newFn (/project/node_modules/babel-traverse/lib/visitors.js:276:21)
   at NodePath._call (/project/node_modules/babel-traverse/lib/path/context.js:76:18)
   at NodePath.call (/project/node_modules/babel-traverse/lib/path/context.js:48:17)
   at NodePath.visit (/project/node_modules/babel-traverse/lib/path/context.js:105:12)
   at convert (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:40:11)
   at converters.TypeAlias (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:304:14)
   at convert (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:38:12)
   at TypeAlias (/project/node_modules/babel-plugin-flow-runtime/lib/transformVisitors.js:69:47)
   at NodePath._call (/project/node_modules/babel-traverse/lib/path/context.js:76:18)
   at NodePath.call (/project/node_modules/babel-traverse/lib/path/context.js:48:17)
   at NodePath.visit (/project/node_modules/babel-traverse/lib/path/context.js:105:12)
   at TraversalContext.visitQueue (/project/node_modules/babel-traverse/lib/context.js:150:16)
   at TraversalContext.visitMultiple (/project/node_modules/babel-traverse/lib/context.js:103:17)
   at TraversalContext.visit (/project/node_modules/babel-traverse/lib/context.js:190:19)
   at Function.traverse.node (/project/node_modules/babel-traverse/lib/index.js:114:17)
   at traverse (/project/node_modules/babel-traverse/lib/index.js:79:12)
   at NodePath.traverse (/project/node_modules/babel-traverse/lib/path/index.js:144:25)
   at PluginPass.Program (/project/node_modules/babel-plugin-flow-runtime/lib/index.js:20:14)
   at newFn (/project/node_modules/babel-traverse/lib/visitors.js:276:21)
   at NodePath._call (/project/node_modules/babel-traverse/lib/path/context.js:76:18)
   at NodePath.call (/project/node_modules/babel-traverse/lib/path/context.js:48:17)
   at NodePath.visit (/project/node_modules/babel-traverse/lib/path/context.js:105:12)
   at TraversalContext.visitQueue (/project/node_modules/babel-traverse/lib/context.js:150:16)
   at TraversalContext.visitSingle (/project/node_modules/babel-traverse/lib/context.js:108:19)
   at TraversalContext.visit (/project/node_modules/babel-traverse/lib/context.js:192:19)
   at Function.traverse.node (/project/node_modules/babel-traverse/lib/index.js:114:17)
   at convert (/project/node_modules/babel-plugin-flow-runtime/lib/convert.js:40:11)
   at TypeAlias (/project/node_modules/babel-plugin-flow-runtime/lib/transformVisitors.js:69:47)
   at NodePath._call (/project/node_modules/babel-traverse/lib/path/context.js:76:18)
   at NodePath.call (/project/node_modules/babel-traverse/lib/path/context.js:48:17)
   at NodePath.visit (/project/node_modules/babel-traverse/lib/path/context.js:105:12)
   at TraversalContext.visitQueue (/project/node_modules/babel-traverse/lib/context.js:150:16)
   at TraversalContext.visitMultiple (/project/node_modules/babel-traverse/lib/context.js:103:17)
   at TraversalContext.visit (/project/node_modules/babel-traverse/lib/context.js:190:19)
   at Function.traverse.node (/project/node_modules/babel-traverse/lib/index.js:114:17)
   at traverse (/project/node_modules/babel-traverse/lib/index.js:79:12)
   at NodePath.traverse (/project/node_modules/babel-traverse/lib/path/index.js:144:25)
   at PluginPass.Program (/project/node_modules/babel-plugin-flow-runtime/lib/index.js:20:14)
   at newFn (/project/node_modules/babel-traverse/lib/visitors.js:276:21)
   at NodePath._call (/project/node_modules/babel-traverse/lib/path/context.js:76:18)
   at NodePath.call (/project/node_modules/babel-traverse/lib/path/context.js:48:17)
   at NodePath.visit (/project/node_modules/babel-traverse/lib/path/context.js:105:12)
   at TraversalContext.visitQueue (/project/node_modules/babel-traverse/lib/context.js:150:16)
   at TraversalContext.visitSingle (/project/node_modules/babel-traverse/lib/context.js:108:19)
   at TraversalContext.visit (/project/node_modules/babel-traverse/lib/context.js:192:19)
   at Function.traverse.node (/project/node_modules/babel-traverse/lib/index.js:114:17)
   at traverse (/project/node_modules/babel-traverse/lib/index.js:79:12)
   at File.transform (/project/node_modules/babel-core/lib/transformation/file/index.js:558:35)
```

https://github.com/codemix/flow-runtime/blob/master/packages/babel-plugin-flow-runtime/src/firstPassVisitors.js

**Comment by phpnode** [comment]
@aretecode please could you post your input? Ideally reduced test case, thanks :)

**Comment by phpnode** [comment]
@aretecode if you're having trouble finding it, it's something of the pattern:

```js
type Foo = {
 bar: () => something; // <--- would really like to know what this something is :)
}
```

**Comment by aretecode** [comment]
@phpnode I will make a test case when I find time which hopefully will be soon

I do have something like this matching that pattern

```js
type s3API = {
  uploadFiles: (
    item: ItemType,
    placeholderIds: Array<BlockType.id>,
    fileList: Array<FileType>,
    onProgress: (progress: number, status: string, file: FileType) => mixed,
    onError: (id: uuid, error: Error) => mixed,
    onFinish: (id: uuid) => mixed,
  ) => Promise,
}
```

**Comment by Dema** [comment]
In my case it crashes on unnamed function export:
```
export default function (a: number, b:number) {
        return a+b;
}
```
Here is the minimal project.
[flow-runtime-bug.zip](https://github.com/codemix/flow-runtime/files/691676/flow-runtime-bug.zip)

**Comment by phpnode** [comment]
@Dema awesome, thank you, will push a fix soon

**Comment by Dema** [comment]
There is one more place where it crashes. If I have

```
 export default class {}
```
it crashes here in the same module:
```
 Class (path: NodePath) {
      if (path.isClassDeclaration()) {
        const {name} = path.node.id;
^^^^^^^^^^^^^^^^^^^^^^
        context.defineValue(name, path.parentPath);
      }
```

**Comment by matthewbpt** [comment]
I get this error too, and I narrowed it down to anywhere where my code does a default export of an anonymous function or class, or directly exports an arrow function. I went through all my code making sure everything has a proper name and now it no longer throws that error during compilation. It's a common pattern for small Stateless React Components to just export an anonymous arrow function directly, so I think this bug should be fixed.
